janusgraph-core/src/main/java/org/janusgraph/graphdb/tinkerpop/JanusGraphBlueprintsGraph.java

```diff
@@ -321,11 +322,13 @@ public boolean isOpen() {
         @Override
         public void close() {
             close(this);
+            txs.remove();
         }
 
         void close(Transaction tx) {
             closeConsumerInternal.get().accept(tx);
             Preconditions.checkState(!tx.isOpen(),"Invalid close behavior configured: Should close transaction. [%s]", closeConsumerInternal);
+            super.close();
```

Comment:
`void close(Transaction tx)` confuses me. Do we need `txs.remove();` here also? Do you know that transactions are passed to this method? Are all those transactions thread local for current thread? If so, I don't understand why we need `closeConsumerInternal.get().accept(tx);` and what is the difference with `close()` method. If we should execute `closeConsumerInternal.get().accept(tx);` for custom transaction (i.e. transaction which isn't a thread local transaction for current thread) then I think we should do the same logic here as in `close()` method except before calling it execute `closeConsumerInternal.get().accept(tx);`. 
If so, I would rewrite those 2 `close` methods to something like:
```
@Override
public void close() {
    super.close();
    txs.remove();
    checkTransactionIsClosed(this);
}

void close(Transaction tx) {
    closeConsumerInternal.get().accept(tx);
    checkTransactionIsClosed(tx);
    close();
}

private void checkTransactionIsClosed(Transaction tx){
    Preconditions.checkState(!tx.isOpen(),"Invalid close behavior configured: Should close transaction. [%s]", closeConsumerInternal);
}
```
But again, it is hard for me to understand why the method `void close(Transaction tx)` even exists.
What do you think about it?